src/commands/deploy/deploy.ts

```diff
@@ -88,6 +95,48 @@ async function deploy(actionContext: IActionContext, arg1: vscode.Uri | string |
     await notifyDeployComplete(context, node, context.workspaceFolder.uri.fsPath);
 }
 
+async function tryToUpdateWorkerProcessTo64BitIfRequired(context: IDeployContext, siteConfig: WebSiteManagementModels.SiteConfigResource, node: SlotTreeItemBase, language: ProjectLanguage): Promise<void> {
+    try {
+        const functionProject: string | undefined = await tryGetFunctionProjectRoot(context.workspaceFolder.uri.fsPath);
+        if (functionProject === undefined) {
+            throw new Error(localize('failedToFindFuncHost', 'Unable to locate Azure Functions project.'));
+        }
+        const projectFiles: string[] = await dotnetUtils.getProjFiles(language, functionProject);
+        if (projectFiles.length === 0) {
+            throw new Error(localize('unableToFindProj', 'Unable to detect project file.'));
+        }
+        const mainProject: string = path.join(functionProject, projectFiles[0]);
+        const platformTarget: string | undefined = await dotnetUtils.tryGetPlatformTarget(mainProject);
+        if (platformTarget === 'x64' && siteConfig.use32BitWorkerProcess === true) {
+            const message: string = localize('overwriteSetting', 'Detected 64 bit project. Update setting in Azure Portal?');
+            const deployAnyway: vscode.MessageItem = { title: localize('deployAnyway', 'Deploy Anyway'), isCloseAffordance: true };
+            const dialogResult: vscode.MessageItem = await ext.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, deployAnyway);
+            if (dialogResult.isCloseAffordance === true) {
+                return;
+            }
+            const config: WebSiteManagementModels.SiteConfigResource = {
+                use32BitWorkerProcess: false
+            };
+            // restart fn app otherwise we get error - Function Host is not running
+            context.stopAppBeforeDeploy = true;
+            await node.root.client.updateConfiguration(config);
+        }
+    } catch (error) {
+        // swallow cancellations
+        const parsedError: IParsedError = parseError(error);
+        if (!parsedError.isUserCancelledError) {
+            return;
+        }
+        // We try out best to update app to 64 bit
+        // If we don't succeed, we shouldn't stop the deployment
```

Comment:
This isn't quite what I meant. If the user explicitly chooses to update the bitness, it's okay if an error stops deployment. They're not blocked, because they can just deploy again and select "Deploy Anyway" the next time.

I suggest you get rid of the entire catch block and instead change the above lines where you `throw new Error` to just `return;`. Those cases _would_ block a user from deploying, which is what we're trying to avoid.

Reply:
Removed code which threw error